### PR TITLE
Using the new bulk delete for hbase truncate.

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

**Comment by garye** [comment]
LGTM

**Comment by sduskis** [comment]
This can only be merged once the new server is deployed.

**Comment by carterpage** [comment]
Integration tests fail for me.

testTruncate(com.google.cloud.bigtable.hbase.TestTruncateTable)  Time elapsed: 1.079 sec  <<< ERROR!
com.google.bigtable.repackaged.io.grpc.StatusRuntimeException: UNIMPLEMENTED: Operation is not implemented, or supported, or enabled.
	at com.google.bigtable.repackaged.io.grpc.Status.asRuntimeException(Status.java:431)
	at com.google.bigtable.repackaged.io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:208)
	at com.google.bigtable.repackaged.io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:141)
	at com.google.bigtable.admin.table.v1.BigtableTableServiceGrpc$BigtableTableServiceBlockingStub.bulkDeleteRows(BigtableTableServiceGrpc.java:349)
	at com.google.cloud.bigtable.grpc.BigtableTableAdminGrpcClient.deleteRowRange(BigtableTableAdminGrpcClient.java:70)
	at org.apache.hadoop.hbase.client.AbstractBigtableAdmin.truncateTable(AbstractBigtableAdmin.java:615)
	at com.google.cloud.bigtable.hbase.TestTruncateTable.testTruncate(TestTruncateTable.java:58)

testTruncateWithSplits(com.google.cloud.bigtable.hbase.TestTruncateTable)  Time elapsed: 1.298 sec  <<< ERROR!
com.google.bigtable.repackaged.io.grpc.StatusRuntimeException: UNIMPLEMENTED: Operation is not implemented, or supported, or enabled.
	at com.google.bigtable.repackaged.io.grpc.Status.asRuntimeException(Status.java:431)
	at com.google.bigtable.repackaged.io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:208)
	at com.google.bigtable.repackaged.io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:141)
	at com.google.bigtable.admin.table.v1.BigtableTableServiceGrpc$BigtableTableServiceBlockingStub.bulkDeleteRows(BigtableTableServiceGrpc.java:349)
	at com.google.cloud.bigtable.grpc.BigtableTableAdminGrpcClient.deleteRowRange(BigtableTableAdminGrpcClient.java:70)
	at org.apache.hadoop.hbase.client.AbstractBigtableAdmin.truncateTable(AbstractBigtableAdmin.java:615)
	at com.google.cloud.bigtable.hbase.TestTruncateTable.testTruncateWithSplits(TestTruncateTable.java:98)

File: bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 1]
```diff
@@ -611,26 +612,10 @@ public boolean isAborted() {
 
   @Override
   public void truncateTable(TableName tableName, boolean preserveSplits) throws IOException {
```
Comment (carterpage) [inline bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Please add a comment that preserveSplits is ignored, and splits will remain in place until Bigtable cleans them up.

File: bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 2]
```diff
@@ -611,26 +612,10 @@ public boolean isAborted() {
...
-
-  private byte[][] getSplits(TableName tableName) throws IOException {
-    try (RegionLocator regionLocator = connection.getRegionLocator(tableName)) {
-      List<HRegionLocation> allRegionLocations = regionLocator.getAllRegionLocations();
-      byte[][] splits = new byte[allRegionLocations.size() - 1][];
-      for (int i = 0; i < allRegionLocations.size() - 1; i++) {
-        splits[i] = allRegionLocations.get(i).getRegionInfo().getEndKey();
-      }
-      return splits;
-    }
```
Comment (carterpage) [inline bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: Log statement, eg:

LOG.info("Truncating all rows from table %s.", tableName);

File: bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java [hunk 3]
```diff
@@ -611,26 +612,10 @@ public boolean isAborted() {
...
+    bigtableTableAdminClient.deleteRowRange(BulkDeleteRowsRequest.newBuilder()
+        .setTableName(options.getClusterName().toTableNameStr(tableName.getNameAsString()))
+        .setDeleteAllDataFromTable(true)
+        .build());
```
Comment (carterpage) [inline bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: } catch (Throwable throwable) {
    throw new IOException(
            String.format(
                    "Failed to truncate table '%s'",
                    tableName.getNameAsString()),
            throwable);
  }

File: bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java (hunk 2 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java]: We only log warnings or unexpected information.  I'd rather not log an explicitly requested action; it seems superfluous.

**Pull request merged**